modules/cassandra/cassandra.go

```diff
@@ -94,21 +153,35 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
 			"CASSANDRA_ENDPOINT_SNITCH": "GossipingPropertyFileSnitch",
 			"CASSANDRA_DC":              "datacenter1",
 		},
-		WaitingFor: wait.ForAll(
-			wait.ForListeningPort(port),
-			wait.ForExec([]string{"cqlsh", "-e", "SELECT bootstrapped FROM system.local"}).WithResponseMatcher(func(body io.Reader) bool {
-				data, _ := io.ReadAll(body)
-				return strings.Contains(string(data), "COMPLETED")
-			}),
-		),
 	}
 
 	genericContainerReq := testcontainers.GenericContainerRequest{
 		ContainerRequest: req,
 		Started:          true,
 	}
 
+	var settings Options
 	for _, opt := range opts {
+		if opt, ok := opt.(Option); ok {
+			if err := opt(&settings); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	exposePort, waitStrategies, tcOpts, err := setupTLS(&settings)
+	if err != nil {
+		return nil, err
+	}
+
+	tcOpts = append(tcOpts, testcontainers.WithExposedPorts(exposePort...))
+	tcOpts = append(tcOpts, testcontainers.WithWaitStrategy(waitStrategies...))
+
+	// Append the customizers passed to the Run function.
+	tcOpts = append(tcOpts, opts...)
+
+	// Apply the testcontainers customizers.
+	for _, opt := range tcOpts {
```

Comment:
suggestion: simplify by passing genericContainerReq into setupTLS and having it do the work, instead of double handling the values.

You could even pass it in to the options so you don't even need the IsTLSEnabled boolean and associated check.

```go
type Option func(*testcontainers.GenericContainerRequest, *Options) error
```